components/insight/SRC/org/openmicroscopy/shoola/agents/fsimporter/util/FileImportComponentI.java

```diff
@@ -0,0 +1,404 @@
+     */
+    public abstract void setLocation(DataObject data, DatasetData dataset,
+            Object refNode);
+
+    /**
+     * Sets the log file annotation.
+     * 
+     * @param data The annotation associated to the file set.
+     * @param id The id of the file set.
+     */
+    public abstract void setImportLogFile(Collection<FileAnnotationData> data,
+            long id);
+
+    /**
+     * Returns the dataset or <code>null</code>.
+     * 
+     * @return See above.
+     */
+    public abstract DatasetData getDataset();
+
+    /**
+     * Returns the object or <code>null</code>.
+     * 
+     * @return See above.
+     */
+    public abstract DataObject getDataObject();
+
+    /**
+     * Returns <code>true</code> if the parent is set.
+     * <code>false</code> otherwise.
+     * 
+     * @return See above.
+     */
+    public abstract boolean hasParent();
+
+    /**
+     * Returns the components displaying the status of an on-going import.
+     * 
+     * @return See above.
+     */
+    public abstract Status getStatus();
+
+    /**
+     * Sets the result of the import.
+     * @param image The image.
+     */
+    public abstract void setStatus(Object image);
+
+    /**
+     * Returns the files that failed to import.
+     * 
+     * @return See above.
+     */
+    public abstract List<FileImportComponentI> getImportErrors();
+
+    /**
+     * Returns the id of the group.
+     * 
+     * @return See above.
+     */
+    public abstract long getGroupID();
+
+    /**
+     * Returns the id of the experimenter.
+     * 
+     * @return See above.
+     */
+    public abstract long getExperimenterID();
+
+    /**
+     * Returns the import error object.
+     * 
+     * @return See above.
+     */
+    public abstract ImportErrorObject getImportErrorObject();
+
+    /**
+     * Returns <code>true</code> if the import has failed, <code>false</code>
+     * otherwise.
+     * 
+     * @return See above.
+     */
+    public abstract boolean hasImportFailed();
+
+    /**
+     * Returns <code>true</code> if it was a failure prior or during the
+     * upload, <code>false</code> otherwise.
+     * 
+     * @return See above.
+     */
+    public abstract boolean hasUploadFailed();
+
+    /**
+     * Returns <code>true</code> if the import has been cancelled,
+     * <code>false</code> otherwise.
+     * 
+     * @return See above.
+     */
+    public abstract boolean isCancelled();
+
+    /**
+     * Returns <code>true</code> if the component has imports to cancel,
+     * <code>false</code> otherwise.
+     * 
+     * @return See above.
+     */
+    public abstract boolean hasImportToCancel();
+
+    /**
+     * Returns <code>true</code> if the file can be re-imported,
+     * <code>false</code> otherwise.
+     * 
+     * @return See above.
+     */
+    public abstract boolean hasFailuresToReimport();
+
+    /**
+     * Returns <code>true</code> if the file can be re-imported,
```

Comment:
very similar to `hasFailuresToReimport()`?

Reply:
That's true, both methods have the same implementation. I remove one of them.

Reply:
Looks like I was wrong, I have to leave both methods in.

Reply:
Maybe Javadoc can explain the difference?